Automattic/wp-calypso
# Performance Profiler: add tracks events

Closes https://github.com/Automattic/dotcom-forge/issues/8811

## Proposed Changes

* add tracks events to various events within the Performance Profiler

## Why are these changes being made?


https://github.com/Automattic/dotcom-forge/issues/8811 

## Testing Instructions



* Open Chrome devtools and in the Console tab, paste this command to enable analytics logging:
```
localStorage.setItem( 'debug', 'calypso:analytics*' );
```
* On the Console tab, set the `verbose` logging level, and add the `calypso_performance_profiler` filter: 
  ![CleanShot 2024-09-10 at 14 49 54@2x](https://github.com/user-attachments/assets/375f93c4-99df-430f-9ce4-e9c40657f989)
* Start a new test by adding the site URL as a parameter, eg.:
```
/speed-test-tool?url=google.com
```
* Check in the logs that the tracks events are fired correctly 

| Action | Tracks event | Payload |
|--------|--------|--------|
| Test started | `calypso_performance_profiler_test_started` | siteUrl |
| Test completed | `calypso_performance_profiler_test_completed` | siteUrl, duration, mobile_score, desktop_score |
| Tab changed | `calypso_performance_profiler_tab_changed` | siteUrl, tab |
| Weekly report button clicked | `calypso_performance_profiler_weekly_report_cta_click` | siteUrl |
| Migration button clicked | `calypso_performance_profiler_migration_banner_cta_click` | siteUrl |
| Recommendation item clicked | `calypso_performance_profiler_insight_click` | siteUrl, key: lighthouse audit ID |

https://github.com/user-attachments/assets/0cadc04d-338a-4101-bfb8-25624a852682




## Pre-merge Checklist
